spec/user_spec.rb

```diff
@@ -0,0 +1,92 @@
+require 'user'
+
+describe User do
+  describe 'self.new_user' do
+    it 'creates a new user on the database and a instance of User' do
+      user = User.new_user('littletrout', 'password1', 'littletrout@notadomain.com')
+
+      stored_data = stored_user(table: 'users', user_id: user.user_id)
+      
+      expect(user).to be_a(User)
+      expect(user.user_id).to eq(stored_data.first['user_id'])
+      expect(user.username).to eq(stored_data.first['username'])
+      expect(user.email).to eq(stored_data.first['email'])
+    end
+
+    it 'encrpyts the password with bcrypt' do
+      expect(BCrypt::Password).to receive(:create).with('password1')
+
+      User.new_user('littletrout', 'password1', 'littletrout@notadomain.com')
+    end
+
+    it 'returns 1 if the password entered is less than 8 characters long' do
+      expect(User.new_user('littletrout', '123456', 'littletrout@notadomain.com')).to eq(1)
+    end
+
+    it 'returns 2 if the username has already been used and is in the database' do
+      expect(User.new_user('captainfrog', 'password1', 'captainfrog@notadomain.com')).to eq(2)
+    end
+    
+    it 'returns 3 if the email has already been used and is in the database' do
+      expect(User.new_user('littlebear', 'password1', 'captainfrog@notadomain.com')).to eq(3)
+    end
+  end
+
+  describe 'self.find' do
```

Comment:
perhaps the find methods, while you currently have more than one for the user, could be specified to what you are finding. i.e. self.find_by_id, to complement your other find method